worlds/tunic/er_scripts.py

```diff
@@ -240,9 +243,78 @@ def pair_portals(world: "TunicWorld") -> Dict[Portal, Portal]:
                     portal_name2 = "Shop Portal"
                 plando_connections.append(PlandoConnection(portal_name1, portal_name2, "both"))
 
+    non_dead_end_regions = set()
+    for region_name, region_info in tunic_er_regions.items():
+        if not region_info.dead_end:
+            non_dead_end_regions.add(region_name)
+        elif region_info.dead_end == 2 and logic_rules:
+            non_dead_end_regions.add(region_name)
+
     if plando_connections:
-        portal_pairs, dependent_regions, dead_ends, two_plus = \
-            create_plando_connections(plando_connections, dependent_regions, dead_ends, two_plus)
+        for connection in plando_connections:
+            p_entrance = connection.entrance
+            p_exit = connection.exit
+
+            if p_entrance.startswith("Shop"):
```

Comment:
Because of all the subtly different Shop-related magic strings, I got lost trying to figure out what this is actually supposed to be matching. I think most of the strings which _start_ with Shop actually got deleted, so is this... just checking for "Shop Portal"? Or just "Shop_"? Or multiple names?

Reply:
This is checking for either Shop or Shop Portal primarily. Basically, this is just making it so you can type in either Shop or Shop Portal and it'll just work. This is because in the spoiler, the shops are called "Shop Portal", but I figure people would just put in Shop in the plando, so I wanted that to work too.

The stuff with Shop_ is for internal destinations. Keeping it brief, Tunic uses a destination and a tag to determine where you're going -- destination is the scene, tag is the portal in that scene. This is shown in the format of destination_tag. So, the _ is there because the tag is empty.

In the Ladder Rando PR, I split destination back up into a destination and a tag (so I could make a destination, scene_tag and a scene, destination_tag, basically to get the vanilla connections).

Reply:
Oh and the point of this specific bit right here is because I'm always making sure the shop is second. Basically just to make it so there's less duplicated code on both the AP side and mod side cause shops are weird.

Reply:
> In the Ladder Rando PR, I split destination back up into a destination and a tag (so I could make a destination, scene_tag and a scene, destination_tag, basically to get the vanilla connections).

Sounds good. In that case I'll worry about code clarity when I get to that PR (if no one else beats me to it).

Reply:
That one has actually gotten a couple reviews at this point :D Hard to say if they went as in depth at trying to understand everything, though, so may still be worth a look? Up to you.